Configure `workflow-run-job-linux` to use sccache-dist build cluster

trxcllnt opened the pull request:
Opening this PR to test using the sccache-dist build cluster in CI.

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

wmaxey commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test https://github.com/NVIDIA/cccl/pull/2672/commits/490335d9ad390381015de4be8482fd942bc06458

trxcllnt commented:
/ok to test

trxcllnt commented:
/ok to test 490335d

trxcllnt commented:
/ok to test e811f3594a60b19e87ed6bf384662ac6549a0979

trxcllnt commented:
/ok to test ef820166074847f9c96972f351d0ffb4415d016b

trxcllnt commented:
/ok to test 79bc40bfe4d2471e96b537466037ab464dcb71d2

trxcllnt commented:
/ok to test afdf41d99a4603899fac817c81984421839507b0

trxcllnt commented:
/ok to test ddbc8c85eb3c4ed48d169c962597006cbd40a895

trxcllnt commented:
/ok to test 73d8e7e9fbe72ca09cf37f86c7c07d4d52c8ec61

trxcllnt commented:
/ok to test e02632f60c35693ced7cf37be33ac5d9b5c1f508

trxcllnt commented:
/ok to test 3611862378ffbf471f8855e6d2fcd83c06ec7655

trxcllnt commented:
/ok to test 292083fb35c99e4d249f6be39b49f509e76619ea

trxcllnt commented:
/ok to test 666172ab08d734f327fb60771c2ec20105644cfc

trxcllnt commented:
/ok to test 2df4f93df9c722f8c1e72119333fbaabc2fae4bc

trxcllnt commented:
/ok to test fdfe4859005cb85cae67ab4932da8b3b8b86ac24

trxcllnt commented:
/ok to test 611cde3ff09d8ebb9a4c66437fc5ecd323934213

trxcllnt commented:
/ok to test 10cd4cc50990dbd67fa57e42ab54cd3f618e8a79

trxcllnt commented:
/ok to test ff96c1710bb0ee0d96f1137f21bd94d6fccb2b7f

trxcllnt commented:
/ok to test 3f40929f8a292bc0636485ae71bdada51b238cd3

trxcllnt commented:
/ok to test f0046de09ae53406438c39c845cac1fa2c66dfc8

trxcllnt commented:
/ok to test 264d324bb4dcaa1b0e41b44bfdeebf7b62264b83

trxcllnt commented:
/ok to test a57f442204afb0a7810bd1b494251d170018dd83

trxcllnt commented:
pre-commit.ci autofix

trxcllnt commented:
/ok to test 13cfa2ef697771c6bb26d84b3d85d10ddaa2e10d

trxcllnt commented:
/ok to test 948a2b38dc5f5019aa6b091e098c8826e25eb835

trxcllnt commented:
/ok to test 04295184ed38dc7ab02eccbe585838f655027fc4

trxcllnt commented:
/ok to test 3fc980c2656e3477a65ef2b00f727fc3ca897c69

trxcllnt commented:
/ok to test 23f917519b725e28e57479bc5ba282da12936da3

trxcllnt commented:
/ok to test b3cc720804a15182b06342faec335ec896cbdb5f

trxcllnt commented:
/ok to test 42d422431a7bf94258d6fe82a5a73a9deb2c7f1c

trxcllnt commented:
/ok to test 71fb197ce3c8cb6b8df965ef9c5a884f53b07bda

trxcllnt commented:
/ok to test 06b3e21e73571bc6d1088e83f1714da12e9de9b4

trxcllnt commented:
/ok to test d170b92dfe69292233e46b02e127bde842f3ed41

trxcllnt commented:
/ok to test 575dcaa7690ce4cf146f2fe5f6af527374babb20

trxcllnt commented:
/ok to test 14f71194e1a6a6ed2e07f2720b227730c3bbfd84

trxcllnt commented:
/ok to test 92f090f7c524056bb7dea0d51bf03cccff56ce8d

trxcllnt commented:
/ok to test c4b660781a47f92f5c8a6e8fed4a7f6be4ed4582

trxcllnt commented:
/ok to test 98913618ceb38c0cd19b388c748e67ae64e337f0

trxcllnt commented:
/ok to test 35070104a93244ed18d8a165568ac5ba82cc27a9

trxcllnt commented:
/ok to test cba277a1234e0729fc08f52b6792f1ad7e25b1bb

trxcllnt commented:
/ok to test 1266fd8f97c88e847f9cf0c6fe7f633fe1822606

trxcllnt commented:
/ok to test 019cbdadddbf301c6e15c85677bc712c66b769f8

trxcllnt commented:
/ok to test f03c68e90a8e0873a9b2782d061414de3481f784

trxcllnt commented:
pre-commit.ci autofix

trxcllnt commented:
/ok to test d153c55a48a8ed9e0f06fe2b4a648022b4ac304a

trxcllnt commented:
/ok to test 43793f1796b768bf98dcb6b950b46a5900969b33

trxcllnt commented:
/ok to test 7aae4d47d82e730a7c47a581fd4454279134ae41

trxcllnt commented:
/ok to test 106806fb95566171cd8308b22faef66bbe4acdf8

trxcllnt commented:
/ok to test 0dc968507227fd66a61c5039a24057c92ef0d899

trxcllnt commented:
/ok to test aa634f2c85f791968bb27690c29c8afe6f9cd163

trxcllnt commented:
/ok to test 13b3f95fca59e4a73528494956930d446078a55d

trxcllnt commented:
/ok to test 93b5f3fd24cc1636102a0509739ea5a12eff09fe

trxcllnt commented:
/ok to test 464fe61425400e32969149c952f0db1d651276c4

trxcllnt commented:
/ok to test 774f4a3b8da9596c43fed9473799f04fddf5d3b1

trxcllnt commented:
/ok to test b4c495fef736d488338f9b122ca34cc7d8a24994

trxcllnt commented:
/ok to test 689d98024533d8b759c0c1f8713b1bae02f40a0a

trxcllnt commented:
/ok to test 3d7a077395395ca630e4293636669db5746d0b54

trxcllnt commented:
/ok to test 21b2305c1cd9ba18047738910a8ee2cb9d29e821

trxcllnt commented:
/ok to test 74c3c5222ac994a9d3f96fc096d4fdda2f93a73d

trxcllnt commented:
/ok to test 59a997c763f48c20e57dcc4b658513f475525149

trxcllnt commented:
/ok to test 12b627aafeaa3909a18880bd1b5195c9a482896e

trxcllnt commented:
/ok to test 8b5d30e8b2370dbebd3b67ca1580889cd85845d6

trxcllnt commented:
/ok to test 8f950f948aac92423448a696c9587a6f2f96518f

trxcllnt commented:
/ok to test f61f8dc9d40ca0d964372a16afcad1b0adb218b9

trxcllnt commented:
/ok to test e89b2b8665ec8b7981f6b1b07bf528069b5f903e

alliepiper committed:
Add NVHPC 25.11/26.1 to CI/devcontainers, 26.04 bump.

alliepiper committed:
temp override matrix.

trxcllnt commented:
/ok to test fa4d5e67bb4a18f18c48bf5d80f9ba58cd7e5232

trxcllnt commented:
/ok to test 31130e67bf5dbe90a91b44cc757a08238e581710

trxcllnt commented:
/ok to test 3eb2ba902915ba3a6a5ef76ac3964fd7eaac9798

trxcllnt commented:
/ok to test f414d9e4daeff11885519f1a5943451a4552643a

trxcllnt commented:
/ok to test e02bb1bd0c76c7b79a80c307d86a82035f092c77

trxcllnt commented:
/ok to test 974db4608cd68defee16c067f89f028fc927a0de

trxcllnt commented:
/ok to test d1bc90accd018131cf8b778fc5f2d5228345ccbf

trxcllnt commented:
/ok to test 3dead4acd9a77f10aeff4a9ad8dc6d0b4dd68cdc

trxcllnt commented:
/ok to test e7cd35783785386c1abbb1b5be13f4d3127b4e91

trxcllnt commented:
pre-commit.ci autofix

miscco commented:
/ok to test [0f0114b](https://github.com/NVIDIA/cccl/pull/2672/commits/0f0114bb286b2fcc5a4daf854f950ed46f0d8b90)

miscco commented on .github/actions/workflow-run-job-linux/action.yml:
```diff
@@ -20,6 +20,9 @@ inputs:
     required: true
+  # This token must have the "read:enterprise" scope
+  dist-token:
+    description: "The token used to authenticate with the sccache-dist build cluster."
```
How can we enforce the  comment?

miscco commented on .github/actions/workflow-run-job-linux/action.yml:
```diff
@@ -150,37 +167,59 @@ runs:
+        #     # Do not cache build products
+        #     # --env "SCCACHE_NO_CACHE=1"
+        #   )
+        # fi
```
Can this be removed?

trxcllnt replied:
I've left this in so it's easy to uncomment for testing.

trxcllnt replied:
Removed

trxcllnt commented:
/ok to test 8dd1894c8bfb4854f09a3e320703451ecc5f8d16

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test c949f213b3cb61f9017713464fb197e01b72b1d2

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
mount in /etc/timezone and /etc/localtime

trxcllnt committed:
don't run launch.sh in subshell to capture errors thrown by entrypoint scripts

trxcllnt committed:
test smaller runners

trxcllnt commented:
/ok to test b6fb0fb0d3bdc95c738537a2f26e45ce8e6e6a2f

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test b6a40da1256d3f9b78e0a1700ca7dd340b4c94ca

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
mount /etc/timezone and /etc/localtime into devcontainers

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fix/mount-etc-time

trxcllnt committed:
pass PARALLEL_LEVEL to cmake --build in ci/build_stdpar.sh

trxcllnt committed:
pass PARALLEL_LEVEL to cmake --build in ci/build_matx.sh

trxcllnt committed:
Disable build caching for CUB compile tests

trxcllnt committed:
use CMAKE_CUDA_ARCHITECTURES as lit `compute_archs` config

trxcllnt committed:
bump ctest timeout to match CI timeout

trxcllnt committed:
stream lit pre-compile output to stdout so it doesn't look like a CI jbb is hung

trxcllnt committed:
Include ninja, ctest, and sccache logs in build artifacts

pre-commit-ci[bot] committed:
[pre-commit.ci] auto code formatting

trxcllnt committed:
retry failed image pulls 10 times

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7482' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7483' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7484' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7485' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7486' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7487' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'pull-request/7488' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
revert changes to libcu++ lit python files

trxcllnt committed:
revert changes to libcu++ test CMakeLists.txt

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Use GHA id-token for sccache-dist auth

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
remove SCCACHE_DIST_AUTH_TOKEN from devcontainer.json files, define it in workflows

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
default to SCCACHE_SERVER_LOG=sccache=trace

trxcllnt committed:
set override in ci/matrix.yaml

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

miscco committed:
Reenable NVHPC in PR ci

We hopefully have addressed most of the flakiness issues in CI, so lets try to reenable

@trxcllnt for visibility

miscco committed:
Fix cudart dependency for nvhpc builds

miscco committed:
Suppress warnings about kernel visibility

miscco committed:
Avoid NVHPC internal compiler error

NVC++-S-0000-Internal compiler error. unrecognized enum base type

miscco committed:
Work around NVHPC compiler bug

miscco committed:
use the override matrix

alliepiper committed:
Update devcontainers to latest, re-enable nvhpc PR jobs.

miscco committed:
Fix compile error for cudax

alliepiper committed:
Remove nvhpc module handling.

This has been moved into the devcontainer config.

alliepiper committed:
Merge branch 'nvhpc_25.11' into reenable_nvhpc_pull_request

davebayer committed:
Add missing `if_consteval_in_nonconstexpr_function` suppression for nvhpc

alliepiper committed:
Merge #7706 for testing.

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
update devcontainer tags to latest, nvhpc 25.11/26.1

trxcllnt committed:
Merge branch 'reenable_nvhpc_pull_request' of github.com:miscco/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test a94303f205edc9faba56a1d5da20420ac36501a9

trxcllnt committed:
fix RAPIDS cmake versions.json

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 05652ccfaf16e2a5c2f237a5f389bfd01f358f93

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
remove override matrix

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
add nvhpc jobs to matrix override

bernhardmgruber committed:
Reduce warpspeed scan to 256 threads/block on NVHPC

Fixes: #7700

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'warpspeed_nvhpc' of github.com:bernhardmgruber/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
sccache=debug

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
revert changes related to nvhpc and thrust

trxcllnt committed:
bump up parallelism

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
fix lint

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use RAPIDS token service to generate S3 creds

trxcllnt committed:
fix envvars

trxcllnt committed:
use uname instead of dpkg

trxcllnt committed:
use sccache-dist build cluster in optional third-party jobs [skip-matrix]

trxcllnt committed:
document --ulimit

trxcllnt committed:
add override_cccl_version input

trxcllnt committed:
pass run_args after RUN_ARGS

trxcllnt committed:
test uncached matx and pytorch builds without build cluster [skip-matrix]

trxcllnt committed:
test uncached matx and pytorch builds with the build cluster [skip-matrix]

trxcllnt committed:
enable xtrace, pass PARALLEL_LEVEL to ninja in pytorch build [skip-matrix] [skip-vdc]

trxcllnt committed:
don't use the build cluster when configuring pytorch [skip-matrix] [skip-vdc]

trxcllnt committed:
report build times [skip-matrix] [skip-vdc]

trxcllnt committed:
don't use the build cluster when configuring matx [skip-matrix] [skip-vdc]

trxcllnt committed:
don't retry failures in matx [skip-matrix] [skip-vdc]

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
remove SCCACHE_RECACHE=1, fallback to building locally after retrying once, get auth token before invoking launch.sh

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
remove xtrace

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster-in-optional-jobs

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'fea/use-sccache-build-cluster-in-optional-jobs' into fea/use-sccache-build-cluster

trxcllnt committed:
don't automatically set SCCACHE_DIST_URL in postCreateCommand

trxcllnt committed:
only use the build cluster for every 20th run

trxcllnt committed:
set -eo pipefail

trxcllnt commented:
/ok to test 3ba57b449471a28ff390bcaca52fe2d4fff4f8c2

trxcllnt committed:
only use additional parallelism in jobs that use the build cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use auth token from env without passing it on the command line

trxcllnt committed:
print helpful message

trxcllnt commented:
/ok to test 42f7c158645062cc227d9137bf4921bf4ea95488

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use random

trxcllnt commented:
/ok to test 91d4778f603c9cb530de4d09908ed053345fa1b6

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use the build cluster for 1/10 jobs

trxcllnt commented:
/ok to test e87c4d53a5988270161ef37570dc4fdd8e9645fd

trxcllnt committed:
update devcontainers

trxcllnt commented:
/ok to test 38d15890608f0d5074bc304c90d26562bfece33e

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 0f6edd005266008568c46d474ebd0ce0fdbbd135

trxcllnt committed:
revert changes to cccl cuda python scripts

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
tune PARALLEL_LEVEL based on free memory

trxcllnt commented:
/ok to test 66835a51d5c72b520514fcbf94aa75717a7fcce1

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
add lines accidentally deleted when reverting other changes

trxcllnt commented:
/ok to test b4d51fe4e2b05de7951a202cf3430b4eae56d043

trxcllnt committed:
fix lint

trxcllnt committed:
source build_common.sh

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
update PARALLEL_LEVEL tuning

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test b1e18e4650617a5402765b089ecd006f706132a2

trxcllnt committed:
update PARALLEL_LEVEL tuning

trxcllnt commented:
/ok to test 7fbe863edf381ff8fd372388f10645d3719aee66

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test e7bf3e85069366d0f94681d13ebb63e33c6b44bb

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
update PARALLEL_LEVEL tuning

trxcllnt commented:
/ok to test 78f80a65ccb9b8fc681483c009872f9a05999052

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use default PARALLEL_LEVEL

trxcllnt commented:
/ok to test ea6388e58743d2da3d959a361b00bc33eda9b358

trxcllnt committed:
don't pass the SCCACHE_DIST_AUTH_TOKEN value in the command line

trxcllnt committed:
determine whether to use sccache-dist in build-workflow.py so we know to use a runner with more memory

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test b9f7571e918939f63cb9724368b59075b211abf9

trxcllnt committed:
install the latest sccache client

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 5bf7efb5b4c246741323e14f5258a1184ba91b99

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test a2c9b23082f61833e831e96088bc39de970ed083

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 3e39ac462ebc84cdbda8dabb538c8b159ab1d7e5

jrhemstad commented on .github/actions/workflow-run-job-linux/action.yml:
```diff
@@ -120,7 +128,10 @@ runs:
         echo -e "\e[1;34m${COMMAND}\e[0m"
         exit_code=0
-        eval "${COMMAND}" || exit_code=$?
+        # kill 10m early so we have time to upload artifacts for debugging hung builds
```
question: Do we still need this? Or was this just from debugging?

trxcllnt replied:
This was so logs would still get uploaded in case of hung jobs, but now I think the `CCCL_CI_COMMAND_TIMEOUT` [here](https://github.com/NVIDIA/cccl/blob/f52338f93af68c3a1219934c45bf3226f5b7614f/ci/build_common.sh#L200-L204) will accomplish the same thing.

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
remove dead code

trxcllnt commented:
/ok to test c12b5b142c341ba1c8666dfe0f2bb7e7548a6547

trxcllnt committed:
remove timeout from .github/actions/workflow-run-job-linux/action.yml

davebayer commented:
/ok to test 5df99ae

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 3fee82629ada74aec77c554edc2970c59b71ba48

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use 16-core runners with more memory

trxcllnt commented:
/ok to test 71c403402d733152744bf56b4fda9f84ad662be8

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 31799584eb90d8b2e69b07eab2af9a9ee51a6d12

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
disable __make_signed and __make_unsigned buitins when using clang and sccache

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 505be083dcb4b14caa8e74932636c3b6a3a609dd

wmaxey commented:
`--parallel 15 -> --parallel 452` Completely sane. LGTM. Approved.

miscco commented on libcudacxx/include/cuda/std/__type_traits/make_unsigned.h:
```diff
@@ -36,6 +36,12 @@
 
+// __make_unsigned in clang doesn't work with sccache-dist
+// see: https://github.com/llvm/llvm-project/issues/206201
+#if defined(__clang__) && defined(__PREPROCESS_ONLY__)
```
Should be `_CCCL_COMPILER(CLANG)`

trxcllnt replied:
Done :+1:

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
__clang__ -> _CCCL_COMPILER(CLANG)

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 5d5376a2dc093408fe3cf86ee5f851e67cf7034f

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
use build cluster for 50% of cudax jobs, 10% of CUB, Thrust, and libcu++ jobs

trxcllnt committed:
cpu16m -> cpu8m

trxcllnt commented:
/ok to test d64b34bb534335492a959eda9ccc6c42d4ad2c6c

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 44c3b68d7b0ded84c8f4e087cd68d6c89337aee0

trxcllnt committed:
create a swapfile

trxcllnt commented:
/ok to test 37c62776b75bf91b4c6785895047ee545c4f651e

trxcllnt committed:
remove xtrace

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
cpu8m -> cpu8

trxcllnt commented:
/ok to test 2d13fac2778f99ca2977a99f894049dbb1918752

trxcllnt committed:
Merge branch 'main' into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'fea/use-sccache-build-cluster' of github.com:trxcllnt/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 07c4f5e1717218d5781a5629da57c1d71a7e2b59

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 9f0c632aef4acf0225967a453f72b38c87154cce

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
remove duplicate run_args

trxcllnt commented:
/ok to test 0d985aa2bcf2d382bd28d9ff7def6c79c5e369d4

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 4d14ed1135d86a7f7341d2281cd6fd4bdcf70c3f

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 99352a1a26322cafc6757a87e6109cc2fab0619e

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 0e7ccb471ab8f8a50f6495c9d14763cfa092ac68

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
add (dist) to job names that use sccache-dist

trxcllnt commented:
/ok to test 4f6372069d20c582b977a5c6e96be114c8cdd5ee

trxcllnt committed:
use sccache-dist in CUB jobs

trxcllnt commented:
/ok to test 1252040beb213de3ab613a3cb2686e4c03f587db

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
cpu8 -> cpu16

trxcllnt commented:
/ok to test a5c88a062a22cbe7c5c478e997faa9b88c5bd2d6

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Use the build cluster for 1 of 10 cudax jobs

trxcllnt committed:
set USE_SCCACHE_DIST=true|false

trxcllnt committed:
enable sccache dist by default for local developers

trxcllnt commented:
/ok to test e6054d316d55e638db937a141b8818b8c0f11631

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test b39b6e8df1649c5935c5fe3083562a3e558be4f1

trxcllnt committed:
don't duplicate jobs that use the build cluster, always use build cluster for producer jobs

trxcllnt commented:
/ok to test 22b84066e353eba894501f4e688884f58b5a7d53

trxcllnt committed:
fix lint

trxcllnt committed:
always define USE_SCCACHE_DIST envvar

trxcllnt commented:
/ok to test 075fae4ffd6130f30a4cb12261ce7f502ad224d8

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
update to RAPIDS v26.10

trxcllnt commented:
/ok to test 3f1d3ef0503f3259a3bb3023b060956d83966dbc

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
build rapidsmpf before cudf

trxcllnt commented:
/ok to test 2e7980f27e9f9e55b94b8df0b84ee513e514362d

trxcllnt committed:
define PARALLEL_LEVEL in workflow-run-job-linux if not defined

trxcllnt committed:
default to as much parallelism as possible

trxcllnt commented:
/ok to test e3741f593120bd206c7a5e51fefd84fc2b1e5d93

trxcllnt committed:
set SCCACHE_DIST_MAX_RETRIES to 4 in CI

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test aefc7b08dff816f4be68a0406c626e6712ebf810

trxcllnt committed:
set SCCACHE_DIST_MAX_RETRIES to 2 in CI and local dev

trxcllnt commented:
/ok to test e70fc95e06f1e3cca2164ccd84542b56b6c13cb0

trxcllnt committed:
fix typo

trxcllnt committed:
use build cluster for all arm64 jobs

trxcllnt commented:
/ok to test ad7abe50e7310566cff1de213efa7534da5297ec

trxcllnt committed:
remove cuopt from list of RAPIDS libs

trxcllnt commented:
/ok to test f2c50b001a6ff523cad63098402d96d8502a972d

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 2d076841b5c8c437c377e0213077a9cc961c47c0

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test cb876d3c16824191d38cb817cc8f596e462667bd

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 3332b5e75fde7102c36516ce081dc3992170c76a

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test e19dcc5d6e68ec23a94456f838ea0629f07ddaa0

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 2fca0c0aebe1a49a94dcf316a23b492f54c9b579

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 8af198f4007cac642b748ac4f2c5139eeb4b0d76

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 7fd07e44e69498b0dcd01565b3fd1160f1b27661

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
set SCCACHE_DIST_REQUEST_TIMEOUT to 30 minutes in CI

trxcllnt commented:
/ok to test 9cfbfd9f569e303e84fa71b071a99b149f2df9c4

trxcllnt committed:
set SCCACHE_DIST_REQUEST_TIMEOUT to 45 minutes in CI

trxcllnt commented:
/ok to test 020b39e450f39051dab75a5a9df8c545022c8769

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 89571410b287c2482aac248b25aa0de4703a45c9

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
remove devcontainer-lock.json

trxcllnt commented:
/ok to test 0cf361aa7d01890dd0ca8d0a3b414c9eac460d31

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test ddb4dbc567cfb579c61e289a7f3ef124a00fce6e

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
update PARALLEL_LEVEL calculation

trxcllnt commented:
/ok to test 6324f484ea1131c184f7afc2de7c0503358c3522

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt commented:
/ok to test 01726a2dacf2bc777572c03c13f82339f05d5efa

trxcllnt committed:
set USE_SCCACHE_DIST=false

trxcllnt commented:
/ok to test bb561ed454ddc4bd88fe340b23e097e4ecc27bb6

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
only modify PARALLEL_LEVEL if USE_SCCACHE_DIST=true

trxcllnt commented:
/ok to test 90694ab6138ff790f611a0962ae9409ed88efec6

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
Merge branch 'main' of github.com:NVIDIA/cccl into fea/use-sccache-build-cluster

trxcllnt committed:
set floor on N_CPUS - 1

trxcllnt commented:
/ok to test 2709481dbddeef8dbc9b1c9e385810c85a9c1c99

trxcllnt committed:
print_environment_details in ci/build_compile_time_bench.sh

trxcllnt commented:
/ok to test e2e7eb28e5d72d57f266d228a9cf86771f5bc9d9

trxcllnt committed:
ensure PARALLEL_LEVEL is never <= 0

trxcllnt commented:
/ok to test b98fb3b30833933b1fc0c9f0ff82def6c9c07151

trxcllnt commented:
/ok to test d577d87da5b59f6c7f145ea804c4faecaac2f3c8

trxcllnt commented:
/ok to test 8387eef733aa5a2833efdcef6edbbec3a1bea522

trxcllnt commented:
/ok to test 837b8676f5a64de6faef765caef0b1d00fc09e6a

trxcllnt commented:
/ok to test d7df7be3407e48b3ae6ea32f7e49d3a6a9c756b7

trxcllnt commented:
/ok to test 0b6e283d85054a72f7186d3a1f3ed60e1873207c

trxcllnt commented:
/ok to test 06acd97e387dea30bbca2541ef81e7a76bc1b53a

trxcllnt commented:
/ok to test 6b319043e72ee1e9fec612b600adcacf43a9a700

trxcllnt commented:
/ok to test bea484aba4d47384b38bfdeaf8549115b99779f5

trxcllnt commented:
/ok to test 1c82d3619b00463f204fb91c226154695868aef9

trxcllnt commented:
/ok to test 4ec3c42cec8e8ff98eeda191b4018a22a078cc8d

trxcllnt commented:
/ok to test 064767f2ad8dce47cf110325008853ccb49eed87

trxcllnt commented:
/ok to test c224bb15e585ac8ce227c13ffac93f7976b108f9

trxcllnt commented:
/ok to test b72a48e354580e657e6886f3e3224900dba7da46

trxcllnt commented:
/ok to test c9eae1da0f2caf631157c1710d368f51ce9603a0

trxcllnt commented:
/ok to test d681c2ff85b7d21f1e358aede0c6d6e067f609af

trxcllnt commented:
/ok to test 33bca58eb55e593494306603ec6a668f83e581d0

trxcllnt commented:
/ok to test 6781e66bce9ef6f45d8ab6657b8a38b1b5bed7eb

trxcllnt commented:
/ok to test e2a01f9dadb0967ff35068194c71562e2d150825

trxcllnt commented:
/ok to test ec112ef9cbd99ddc6163898e7228098a73906e16

trxcllnt commented:
/ok to test 00fc55f57d9ca00468ff14d74610f8bd4bf4a74e

trxcllnt commented:
/ok to test 2329c7ae3dce0d0bd607770dca5271bf87407384

trxcllnt commented:
/ok to test 4d2b1506ea523a45f74c49992f8d281c17713121

trxcllnt commented:
/ok to test bbdf10c5573c322009435d2e05ebce07ab9e3bd3

trxcllnt commented:
/ok to test f1399393b50ca260579c2c100af61e9dfe42515d

trxcllnt commented:
/ok to test 4e883eb89feb67ba3d478b7122c4ff752eb4ccc3

trxcllnt commented:
/ok to test aa4e27ed40a5ac711da7683b3158f613598195b3

trxcllnt commented:
/ok to test 98e92b2c873863458205e4e82ff3c07ef76a901a

trxcllnt commented:
/ok to test 36a3d9fe72f83be00fa796a8b3f49bbcda596946

trxcllnt commented:
/ok to test 0876185b3ab2ef7a075d72678368d2d40f863647
